addons/mail/static/src/messaging/component/notification_list/notification_list_tests.js

```diff
@@ -0,0 +1,216 @@
+
+        /**
+         * @param {Object} param0
+         * @param {string} [param0.filter='all']
+         */
+        this.createNotificationListComponent = async ({ filter = 'all' }) => {
+            NotificationList.env = this.env;
+            this.notificationList = new NotificationList(null, { filter });
+            await this.notificationList.mount(this.widget.el);
+            await afterNextRender();
+        };
+
+        this.start = async params => {
+            let { env, widget } = await utilsStart(Object.assign({}, params, {
+                data: this.data,
+            }));
+            this.env = env;
+            this.widget = widget;
+        };
+    },
+    afterEach() {
+        utilsAfterEach(this);
+        if (this.notificationList) {
+            this.notificationList.destroy();
+        }
+        if (this.widget) {
+            this.widget.destroy();
+        }
+        this.env = undefined;
+        delete NotificationList.env;
+    }
+});
+
+QUnit.test('notifications are ultimately ordered by date', async function (assert) {
+    assert.expect(3);
+
+    Object.assign(this.data.initMessaging, {
+        channel_slots: {
+            channel_channel: [{
+                channel_type: 'channel',
+                id: 100,
+                name: "Channel 2019",
+                message_unread_counter: 0,
+            }, {
+                channel_type: 'channel',
+                id: 200,
+                name: "Channel 2020",
+                message_unread_counter: 0,
+            }],
+        },
+    });
+    await this.start({
+        async mockRPC(route, args) {
+            if (args.method === 'channel_fetch_preview') {
+                return [
+                    {
+                        id: 100,
+                        last_message: {
+                            author_id: [10, "Author A"],
+                            body: "<p>Message A</p>",
+                            channel_ids: [100],
+                            date: "2019-01-01 00:00:00",
+                            id: 42,
+                            message_type: 'comment',
+                            model: 'mail.channel',
+                            record_name: 'Channel 2019',
+                            res_id: 100,
+                        },
+                    },
+                    {
+                        id: 200,
+                        last_message: {
+                            author_id: [20, "Author B"],
+                            body: "<p>Message B</p>",
+                            channel_ids: [200],
+                            date: "2020-01-01 00:00:00",
+                            id: 43,
+                            message_type: 'comment',
+                            model: 'mail.channel',
+                            record_name: 'Channel 2020',
+                            res_id: 200,
+                        },
+                    }
+                ];
+            }
```

Comment:
There's already a default mock for channel_fetch_preview (see `mail.MockServer:_mockChannelFetchPreview()`).
So I prefer putting messages in data, so that test relies on default mocked rpc.